File: fluentcore/src/main/java/com/mcreater/fxfluent/syslib/Win32UiShellLib.java
package com.mcreater.fxfluent.syslib;

import com.sun.jna.Library;
import com.sun.jna.Native;
import com.sun.jna.platform.win32.WinDef;

public interface Win32UiShellLib extends Library {
    Win32UiShellLib INSTANCE = Native.load("UiShell", Win32UiShellLib.class);
    void NativeWarningReflect();
    WinDef.DWORD GetCompositionColor();
    WinDef.BOOL GetThemeIsDark();
    WinDef.BOOL GetTransparencyEnabled();
    WinDef.BOOL ApplyBlur(WinDef.HWND hwnd, WinDef.DWORD blurType, WinDef.BOOL isDark);
    WinDef.BOOL IsWindows11();
    void SetWindowIsDark(WinDef.HWND hwnd, WinDef.BOOL isDark);
    WinDef.BOOL SetWindowRadius(WinDef.HWND hwnd, WinDef.DWORD radius);
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/syslib/Win32UiShellLib.kt
package com.mcreater.fxfluent.syslib

import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.platform.win32.WinDef.*

interface Win32UiShellLib : Library {
    companion object {
        @JvmField
        val INSTANCE: Win32UiShellLib = Native.load("UiShell", Win32UiShellLib::class.java)
    }
    fun implNativeWarning()
    fun implGetCompositionColor(): DWORD
    fun implGetThemeIsDark(): BOOL
    fun implGetTransparencyEnabled(): BOOL
    fun implApplyBlur(hwnd: HWND?, blurType: DWORD?, isDark: BOOL?): BOOL
    fun implIsWindows11(): BOOL
    fun implSetWindowIsDark(hwnd: HWND?, isDark: BOOL?)
    fun implSetWindowRadius(hwnd: HWND?, radius: DWORD?): BOOL
}